Write data

zkamvar opened the pull request:
This partially addresses #9 by adding a function that will create a table or list of tables as output. @daattali, @kplaney, @thibautjombart, what do you think? 
## NEW FUNCTIONS
- `tablesFromMatch()` generates a tidy table or a list of tables containing the matches
- `getIndexList()` switches the weight output of `matchEpiData()` to index output

The function to generate the table(s) will only output the information used for matching. This way, the user can use it as a reference to filter their own data. When the output is a single tidy table, the information of the group, data set, index, and weight/score are included.
## SHINY

I've modified server.R slightly to use the `giveWeights = TRUE` argument for `matchEpiData()`. This will allow us to write something to display the weights/scores in the future.

zkamvar committed:
Create internal function for weight/index switch

zkamvar committed:
Add collapseValues internal function

zkamvar committed:
Add tableFromMatch function

I've added a function to return tables from the
match query. It can work in addition or in lieu
of `matchEpiData()`. I have added documentation
for this and also updated the documentation of
`matchEpiData()`.

zkamvar committed:
update docs to include library call

zkamvar committed:
add collapse argument to output single data frame

zkamvar committed:
update documentation

zkamvar committed:
add notes to future self and others

zkamvar committed:
Add argument to return thresholds in tidy data

zkamvar committed:
make getIndexList a user-facing function

zkamvar committed:
export getIndexList

zkamvar committed:
update shiny server to conserve the weights

This will allow future development so that we can
give the user the scores/weights in the table that
is output.

daattali commented:
Nice work! I don't think you need to add all the `library(epimatch)` to the examples sections, I don't commonly see that in Hadley et al's packages.  

Regarding `tablesFromMatch`: this is excellent. I haven't looked into it for more than a minute, but does it differentiate between which rows are from which dataset?  In the shiny app, when there's a table showing records from two different linelists, it says which records are from which dataset

zkamvar commented:
I think I put the `library(epimatch)` calls in there during a layover because I was being silly. I can take them out if you like.

`tablesFromMatch()` includes a column the specifies data set as well as their respective indices.

daattali commented:
Oh I see, really nice function. To be nitpicky:
- are the row names in the output needed? If they _are_ needed, would it not make more sense to have them as a proper variable rather than rownames?
- I do see now that using collapse=T prints out where each record came from. But when collapse=F, I don't think there's any identifying information? 

Other than that , this is awesome

zkamvar commented:
> Oh I see, really nice function. To be nitpicky:

Nits are meant to be picked!

> are the row names in the output needed? If they are needed, would it not make more sense to have them as a proper variable rather than rownames?

They are not needed. I can easily get rid of them.

> I do see now that using collapse=T prints out where each record came from. But when collapse=F, I don't think there's any identifying information?

`collapse = FALSE` does not identify the origin (except by rownames, which is a no-no). Should I add that as an extra column?

daattali commented:
It seems to me that it would be more consistent behaviour and make sense to
the end user if the dataset of origin on each record was mentioned. Whether
the result is multiple tables separately or collapsed should probably not
change the amount of output information. If that makes sense to you? Just
my initial thought, but obviously you spent a lot more time with these
functions and data so it's your call! But regarding removing the rownames,
that I do feel strongly about if they're not meaningful :)

---

http://deanattali.com

On 5 August 2016 at 19:01, Zhian N. Kamvar notifications@github.com wrote:

> Oh I see, really nice function. To be nitpicky:
> 
> Nits are meant to be picked!
> 
> are the row names in the output needed? If they are needed, would it not
> make more sense to have them as a proper variable rather than rownames?
> 
> They are not needed. I can easily get rid of them.
> 
> I do see now that using collapse=T prints out where each record came from.
> But when collapse=F, I don't think there's any identifying information?
> 
> collapse = FALSE does not identify the origin (except by rownames, which
> is a no-no). Should I add that as an extra column?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/Hackout3/epimatch/pull/12#issuecomment-238000116, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AA6IFHP5dpBnilaCrgukZwAbnYuK9n5sks5qc-sWgaJpZM4JdJBM
> .

zkamvar committed:
Remove library calls from functions

zkamvar committed:
update Rproj specifications

zkamvar committed:
Add information to list of tables

This addresses Dean's comment:
https://github.com/Hackout3/epimatch/pull/12#issuecomment-238000614

zkamvar committed:
Remove rownames and rename groups to group

This addresses Dean's comment:
https://github.com/Hackout3/epimatch/pull/12#issuecomment-237997712

zkamvar committed:
Add tests for tablesFromMatch
